shopcart-backend/src/main/java/com/shopcart/backend/entity/Order.java

```diff
@@ -0,0 +1,103 @@
+package com.shopcart.backend.entity;
+
+import java.sql.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.persistence.CollectionTable;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MapKeyColumn;
+import javax.persistence.Table;
+import javax.validation.constraints.NotBlank;
+
+import org.hibernate.annotations.CreationTimestamp;
+
+@Entity
+@Table(name="orders")
+public class Order {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+    @Column(nullable = false, updatable = false)
+    @CreationTimestamp
+    private Date date;
+    @Column(name="total")
+    private @NotBlank float total;
+    @Column(name="first_name")
+    private @NotBlank String first_name;
+    @Column(name="last_name")
+    private String last_name;
+    @Column(name="country")
+    private String country;
+    @ElementCollection
+    @CollectionTable(name = "order_products")
+    @MapKeyColumn(name="product_id")
+    @Column(name="quantity")
+    Map<Long, Integer> products = new HashMap<Long, Integer>();
```

Comment:
I think this could be a list of products along with a quantity. A custom object is probably needed called OrderItem with two fields. The product Id and a quantity. We also need checks in the service to ensure demanded quantity does not exceed quantity stored.

Reply:
@ReubenMathew wouldn't checks already be done on the frontend side? As we pass them the product object with how much stock we have (product quantity field), they can just restrict the user from adding more than possible

Reply:
@ReubenMathew also I have attempted to add a a custom object for this but it seems like it adds way too much overhead to the order because it requires me to define **OrderItems** as an entity having its own table and then we own a list of all of its **OrderItem ids** so we would have to create the **OrderItem** first before using it in a order request. We cant reference the whole orderItem in a list like this List<OrderItem> because of the inability of planetscale to add foreign key constraints. Should we still proceed with having an custom object here?

Reply:
Nah I think the custom object isn't needed, I think we're good bro. Thanks.